LSM303AGR.c

```diff
@@ -159,6 +161,88 @@ int32_t lsm303_acc_stop(void)
     return writeReg(LSM303_ACCEL, ACC_CTRL1, reg1);
 }
 
+int32_t lsm303_acc_setINT2(ACC_INT2_type_t mode, uint8_t threshold, uint8_t duration) //type, threshold, time
+{
+	int32_t err;        // error return for the function
+	uint8_t reg5;
+	uint8_t reg6;
+	uint8_t thres_reg;       // ACC threshold setting register 
+	uint8_t cfg_reg;       // ACC INT2 control register 
+	int j;
+	
+	switch(currentScale) {
+		case ACC_FS_2G:  j = 0; 
+		break;
+		case ACC_FS_4G:  j = 1;
+		break;
+		case ACC_FS_8G:  j = 2;
+		break;
+		case ACC_FS_16G: j = 3;
+		break;
+		default: j = 0;
+	};
+	
+	err = readReg(LSM303_ACCEL, ACC_CTRL5, &reg5);
+	err = readReg(LSM303_ACCEL, ACC_CTRL6, &reg6);
+	
+	reg6 |= ACC_CTRL6_I2_INT2;
+	thres_reg = ((threshold) >> j); 
+	cfg_reg = (ACC_INT2_6DMOVE_XYZ); 
+	
+	if(mode == ACC_INT2_6D_en)
+	{
+		reg5 &= ~(ACC_INT2_4D_en);
+	}
+	else if(mode == ACC_INT2_4D_en){
+		reg5 |= (ACC_INT2_4D_en);
+	}
+	
+	err = writeReg(LSM303_ACCEL, ACC_INT2_THS, thres_reg); //Set threshold for interrupt 2
+	err = writeReg(LSM303_ACCEL, ACC_INT2_CFG, cfg_reg); //Enable 6D modes, all the axes and interrupt modes
+	err = writeReg(LSM303_ACCEL, ACC_INT2_DUR, duration); //Set the minimum duration of the Interrupt 2 event to be recognized
+	err = writeReg(LSM303_ACCEL, ACC_CTRL5, reg5);
+	err = writeReg(LSM303_ACCEL, ACC_CTRL6, reg6);
+	
+	if(err != ERR_NONE) { return err; }
+
+	return err;
+}
+
+
+int32_t lsm303_motion_detect(uint32_t* reg_detect)
+{
+	int32_t err;        // error return for the function
+	uint8_t int2_src = 0x00;
+
+	err = readReg(LSM303_ACCEL, ACC_CTRL1, &int2_src);
+	
+	if(int2_src & ACC_INTSRC_IA){
```

Comment:
- Are all of these x > threshold? if not make the comments match or remove them
- fix the indentation. else statements should be on their own line and aligned with the associated if statement.
- This could be a lot easier with some bit shifting.